src/connection/ConnectionOptionsReader.ts

```diff
@@ -94,7 +94,7 @@ export class ConnectionOptionsReader {
         // if .env file found then load all its variables into process.env using dotenv package
         if (foundFileFormat === "env") {
             const dotenv = PlatformTools.load("dotenv");
-            dotenv.config({ path: this.baseFilePath });
+            dotenv.config({ path: `${this.baseFilePath}.env` });
```

Comment:
@steevsachs you're right, but this was not falling for this case if i had `.env` file instead of `ormconfig.env`  What about the `else` case at https://github.com/typeorm/typeorm/pull/3361/files/d0e8fa3af49052cd13703f27f7c31e2d6e5d9d5e#diff-fd8d31a6463f188113f6a9f43f77612fR98 ?

Reply:
The else condition only fires if the config file format is _not_ `.env`, ie `ormconfig.json`.  This ensure environment is still loaded in those cases.

I believe the reason `.env` is working is that
* `ConnectionOptionsReader` uses `ormconfig` as its base file name...
* `foundFileFormat` will equal `undefined` at line 95 because there is no `ormconfig.*` at the base file path
* Thus the `else` condition fires and loads the `.env` file.